### PR TITLE
Removing Compatability Field

### PR DESCRIPTION
**Pull request by AsherGlick**
This field will be replaced with another solution that is centered around each product. One file for each product and in the file all of the support will be listed. This will be a better experience then every product trying to edit a field across multiple files.

### CODE REVIEW

**Review by klingbolt — APPROVED** [review]

File: xml_converter/generators/code_generator.py [hunk 1]
```diff
@@ -589,7 +582,7 @@ def write_webdocs(self, output_directory: str) -> None:
             for field_row in field_rows:
                 complete_field_row_list.append(field_row)
 
-            with open(os.path.join(output_directory, page + "_gen.html"), 'w') as f:
+            with open(os.path.join(output_directory, page + ".html"), 'w') as f:
```
Comment (klingbolt) [inline xml_converter/generators/code_generator.py:585]: What the rational for removing "_gen" from a generated file?

Comment (klingbolt) [inline xml_converter/generators/code_generator.py:585]: What the rational for removing "_gen" from a generated file?

File: xml_converter/generators/code_generator.py [hunk 2]
```diff
@@ -208,7 +201,7 @@ class Generator:
     def delete_generated_docs(self, dir_path: str) -> None:
         for filepath in os.listdir(dir_path):
             filepath = os.path.join(dir_path, filepath)
-            if filepath.endswith("_gen.hpp") or filepath.endswith("_gen.cpp") or filepath.endswith("_gen.html"):
+            if filepath.endswith("_gen.hpp") or filepath.endswith("_gen.cpp") or filepath.endswith(".html"):
```
Comment (klingbolt) [inline xml_converter/generators/code_generator.py:204]: What the rational for removing "_gen" from a generated file?

**Review by klingbolt — CHANGES_REQUESTED** [review]

File: xml_converter/generators/code_generator.py (hunk 1 above)
Comment (AsherGlick) [reply xml_converter/generators/code_generator.py:585]: The _gen suffix was never tested with the html directory and broke all the hyperlinks in the documentation pages when it was added. This fixes them at the expense of the suffix. But the entire html structure is generated, there are no manual parts, so there is no need to delineate between manually created files and the generated files when deleting them as is necessary for the code files. This was shown to be sufficient back before the _gen suffix was added and the html documentation was the only thing being generated.

File: xml_converter/generators/code_generator.py (hunk 2 above)
Comment (AsherGlick) [reply xml_converter/generators/code_generator.py:204]: Responded in the other comment to answer the same question.

**Review by klingbolt — APPROVED** [review]

**Pull request merged**